### PR TITLE
Bounty permissions fix

### PR DESCRIPTION
**Pull request by motechFR**
This PR simplifies the bounty permissions decisions in the UI.

It also re-establishes the ability for a reviewer to submit their work.

Finally, it provides the application editor form outside of the submissions table, so that logic for applying / submitting is separate.

### CODE REVIEW

File: prisma/migrations/20220803193018_drop_bounty_delete_operation/migration.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  - The values [delete] on the enum `BountyOperation` will be removed. If these variants are still used in the database, this will fail.
+
+*/
+-- AlterEnum
+BEGIN;
+CREATE TYPE "BountyOperation_new" AS ENUM ('work', 'review', 'lock', 'approve_applications', 'grant_permissions');
+ALTER TABLE "BountyPermission" ALTER COLUMN "operations" TYPE "BountyOperation_new"[] USING ("operations"::text::"BountyOperation_new"[]);
+ALTER TYPE "BountyOperation" RENAME TO "BountyOperation_old";
+ALTER TYPE "BountyOperation_new" RENAME TO "BountyOperation";
+DROP TYPE "BountyOperation_old";
```
Comment (mattcasey) [inline prisma/migrations/20220803193018_drop_bounty_delete_operation/migration.sql:13]: interesting approach to migration by prisma :)

File: components/[pageId]/DocumentPage/components/BountyProperties.tsx [hunk 2]
```diff
@@ -424,18 +434,30 @@ export default function BountyProperties (props: {children: ReactNode, readOnly?
...
+          <BountyApplicantForm
+            bounty={bounty}
+            submissions={applications}
+            permissions={permissions}
+            refreshSubmissions={refreshSubmissions}
+          />
+        )
+      }
+
       {permissions && bounty.status !== 'suggestion' && (
```
Comment (mattcasey) [inline components/[pageId]/DocumentPage/components/BountyProperties.tsx:461]: I think you should just hide BountySubmissionsTable unless it's a reviewer. There's no reason to show someone the count:
image

File: components/bounties/components/BountyApplicantForm/BountyApplicantForm.tsx [hunk 3]
```diff
@@ -82,20 +83,46 @@ export default function BountyApplicantForm (props: BountyApplicationFormProps)
...
         />
-      );
-    }
+      )
+    );
+  }
+  else {
+
+    const expanded = userApplication.status === 'applied' ? 'application' : 'submission';
+    const showSubmission = userApplication.status !== 'applied';
```
Comment (mattcasey) [inline components/bounties/components/BountyApplicantForm/BountyApplicantForm.tsx:92]: i dont think these two vars are used

**Review by mattcasey — APPROVED** [review]

**Comment by motechFR** [comment]
Outstanding improvements:
- Collapse the submission on submit
- Change submit button from "submit" to "update" once we have submission data
- Hide BountySubmissions table from applicant (so the badge doesn't show)

**Pull request merged**